source/dofs/dof_handler.cc

```diff
@@ -3044,6 +3041,8 @@ DoFHandler<dim, spacedim>::connect_to_triangulation_signals()
   // connect functions to signals of the underlying triangulation
   this->tria_listeners.push_back(this->tria->signals.create.connect(
     [this]() { this->reinit(*(this->tria)); }));
+  this->tria_listeners.push_back(
+    this->tria->signals.clear.connect([this]() { this->clear(); }));
```

Comment:
This looks correct to me. Nothing good can happen if the triangulation is cleared and we still have dangling data.

Reply:
I continue to think that this is correct as is. If you want, break this uncontroversial part out into a patch of its own. Let's get the easy pieces out of the way already.